### PR TITLE
Twitch chat plugin

### PR DESCRIPTION
**Pull request by Adam-**

### CODE REVIEW

**Comment by Adam-** [comment]
@CustardGod can you look at this instead of #2804?

File: runelite-client/src/main/java/net/runelite/client/plugins/twitch/irc/TwitchIRCClient.java [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+		register(username, password);
+		join(channel);
+
+		try
+		{
+			String line;
+
+			while ((line = read()) != null)
+			{
+				log.info("<- {}", line);
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/twitch/irc/TwitchIRCClient.java]: this should probably be log.debug

File: runelite-client/src/main/java/net/runelite/client/plugins/twitch/irc/TwitchIRCClient.java [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+		}
+
+		if (!pingSent && System.currentTimeMillis() - last >= PING_TIMEOUT)
+		{
+			ping("twitch");
+			pingSent = true;
+		}
+		else if (pingSent)
+		{
+			log.info("Ping timeout, disconnecting.");
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/twitch/irc/TwitchIRCClient.java]: Same here, debug

File: runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+			twitchIRCClient.start();
+		}
+	}
+
+	@Schedule(period = 30, unit = ChronoUnit.SECONDS, asynchronous = true)
+	public void checkClient()
+	{
+		if (twitchIRCClient != null)
+		{
+			if (twitchIRCClient.isConnected())
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java]: you can probably move this to l130 else statement

**Pull request merged**